packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -63,32 +63,7 @@ export class AwsS3Publish implements PublisherBase {
     const credentialsConfig = config.getOptionalConfig(
       'techdocs.publisher.awsS3.credentials',
     );
-    let accessKeyId = undefined;
-    let secretAccessKey = undefined;
-    let credentials: Credentials | CredentialsOptions | undefined = undefined;
-    if (credentialsConfig) {
-      const roleArn = credentialsConfig.getOptionalString('roleArn');
-      if (roleArn && aws.config.credentials instanceof Credentials) {
```

Comment:
The old code insists on using AWS.config.credentials. Most default initialisations of the AWS SDK won't set any (using instance profile or the like), and so Role ARNs would have been ignored, despite the instance profile allowing the Role ARN being assumed